File: Algorithms/119_PascalsTriangleII.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> result;
        if (rowIndex == 0) {
            result.push_back(1);
            return result;
        }
        result.push_back(1);
        result.push_back(1);
        for (int i=2; i <= rowIndex; i++) {
            vector<int> v(result);
            result.clear();
            result.push_back(1);
            for (int j=1; j < i; j++) {
                result.push_back(v[j-1]+v[j]);
            }
            result.push_back(1);
        }
        return result;
    }
};

File: Algorithms/119.PascalsTriangleII.java
public class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new ArrayList<>();
        if (rowIndex == 0) {
            result.add(1);
            return result;
        }
        result.add(1);
        result.add(1);
        for (int i=2; i <= rowIndex; i++) {
            List<Integer> v = new ArrayList<>(result);
            result.clear();
            result.add(1);
            for (int j=1; j < i; j++) {
                result.add(v.get(j-1) + v.get(j));
            }
            result.add(1);
        }
        return result;
    }
}
